ebpf/event_common.go

```diff
@@ -87,3 +89,25 @@ func (c ConnectionStats) ByteKey(buffer *bytes.Buffer) ([]byte, error) {
 	}
 	return buffer.Bytes(), nil
 }
+
+func removeDuplicates(conns []ConnectionStats) []ConnectionStats {
+	connections := make([]ConnectionStats, 0, len(conns))
+	seen := map[string]struct{}{}
+	buf := &bytes.Buffer{}
+	for _, c := range conns {
+		key, err := c.ByteKey(buf)
+		if err != nil {
+			// Skip the connection
+			log.Errorf("could not get byte key for connection %v: %s", c, err)
+			continue
+		}
+
+		// If it's the first time we see this connection add it
+		if _, ok := seen[string(key)]; !ok {
```

Comment:
Will taking the `string()` of a `[]byte` give you a correct, consistent key?

Reply:
Yes :)

There is also an optimization to get rid of an unneeded allocation in these cases: https://github.com/golang/go/issues/3512